source/gui/settingsDialogs.py

```diff
@@ -2272,6 +2272,36 @@ def makeSettings(self, settingsSizer):
 			config.conf["virtualBuffers"]["autoFocusFocusableElements"]
 		)
 
+		# Translators: This is the label for a textfield in the
+		# browse mode settings panel.
+		textParagraphRegexLabelText = _("Regular expression for text paragraph navigation")
+		self.textParagraphRegexEdit = sHelper.addLabeledControl(
+			textParagraphRegexLabelText,
+			wxCtrlClass=wx.TextCtrl,
+			size=(self.scaleSize(300), -1),
+		)
+		self.textParagraphRegexEdit.SetValue(config.conf["virtualBuffers"]["textParagraphRegex"])
+		self.bindHelpEvent("BrowseModeSettingsTextParagraphRegexEdit ", self.textParagraphRegexEdit)
+
+	def isValid(self) -> bool:
+		regex = self.textParagraphRegexEdit .GetValue()
+		try:
+			re.compile(regex, re.UNICODE)
+		except re.error as e:
+			log.debugWarning("Failed to compile text paragraph regex", exc_info=True)
+			gui.messageBox(
+				# Translators: Message shown when invalid text paragraph regex entered
+				_("Failed to compile text paragraph regular expression: %s") % str(e),
+				# Translators: The title of the message box
+				_("Error"),
+				wx.OK | wx.ICON_ERROR,
+				self,
+			)
+			self.textParagraphRegexEdit.SetFocus()

```

Comment:
Setting focus when validating should be avoided, as it behaves unexpectedly when applying settings for a different panel (see #15894 for more discussion).